quinn/transformations.py

```diff
@@ -87,28 +88,193 @@ def to_snake_case(s: str) -> str:
     return s.lower().replace(" ", "_")
 
 
-def sort_columns(df: DataFrame, sort_order: str) -> DataFrame:
+def sort_columns(
+    df: DataFrame, sort_order: str, sort_nested: bool = False
+) -> DataFrame:
     """This function sorts the columns of a given DataFrame based on a given sort
     order. The ``sort_order`` parameter can either be ``asc`` or ``desc``, which correspond to
     ascending and descending order, respectively. If any other value is provided for
     the ``sort_order`` parameter, a ``ValueError`` will be raised.
 
     :param df: A DataFrame
-    :type df: pandas.DataFrame
+    :type df: pyspark.sql.DataFrame
     :param sort_order: The order in which to sort the columns in the DataFrame
     :type sort_order: str
+    :param sort_nested: Whether to sort nested structs or not. Defaults to false.
+    :type sort_nested: bool
     :return: A DataFrame with the columns sorted in the chosen order
-    :rtype: pandas.DataFrame
+    :rtype: pyspark.sql.DataFrame
     """
-    sorted_col_names = None
-    if sort_order == "asc":
-        sorted_col_names = sorted(df.columns)
-    elif sort_order == "desc":
-        sorted_col_names = sorted(df.columns, reverse=True)
-    else:
-        raise ValueError(
-            "['asc', 'desc'] are the only valid sort orders and you entered a sort order of '{sort_order}'".format(
-                sort_order=sort_order
+
+    def parse_sort_order(sort_order: str) -> bool:
```

Comment:
This function is called once. Do we really need it? Because there are already so much nested functions... Maybe let's simplify the code a little and replace functions called once by just expressions?